### PR TITLE
Bump versions, expose updateTrip() and getMatrix() (WIP)

### PR DESCRIPTION
**Pull request by nickpatrick**

### CODE REVIEW

File: src/android/src/main/java/io/radar/cordova/RadarCordovaPlugin.java [hunk 1]
```diff
@@ -773,6 +808,66 @@ public void onComplete(Radar.RadarStatus status, RadarRoutes routes) {
...
+            mode = Radar.RadarRouteMode.TRUCK;
+        } else if (modeStr.equals("MOTORBIKE") || modeStr.equals("motorbike")) {
+            mode = Radar.RadarRouteMode.MOTORBIKE;
+        }
+        String unitsStr = optionsObj.getString("units");
+        Radar.RadarRouteUnits units = unitsStr.equals("METRIC") || unitsStr.equals("metric") ? Radar.RadarRouteUnits.METRIC : Radar.RadarRouteUnits.IMPERIAL;
+
+        Radar.getMatrix(origins, destinations, mode, units, new Radar.RadarMatrixCallback() {
+            @Override
+            public void onComplete(Radar.RadarStatus status, RadarRouteMatrix routes) {
```
Comment (brettguenther) [inline src/android/src/main/java/io/radar/cordova/RadarCordovaPlugin.java:854]: this should be defined as `matrix` for RadarRouteMatrix, right?

File: src/android/src/main/java/io/radar/cordova/RadarCordovaPlugin.java [hunk 2]
```diff
@@ -773,6 +808,66 @@ public void onComplete(Radar.RadarStatus status, RadarRoutes routes) {
...
+    public void getMatrix(final JSONArray args, final CallbackContext callbackContext) throws JSONException {
+        final JSONObject optionsObj = args.getJSONObject(0);
+
+        JSONArray originsArr = optionsObj.getJSONArray("origins");
+        Location[] origins = new Location[originsArr.length()];
+        for (int i = 0; i < originsArr.length(); i++) {
+            JSONObject originObj = originsArr.getJSONObject(i);
+            double originLatitude = originObj.getDouble("latitude");
+            double originLongitude = originObj.getDouble("longitude");
+            origin = new Location("RadarCordovaPlugin");
```
Comment (brettguenther) [inline src/android/src/main/java/io/radar/cordova/RadarCordovaPlugin.java:820]: should be `Location origin =...`

File: src/android/src/main/java/io/radar/cordova/RadarCordovaPlugin.java [hunk 3]
```diff
@@ -87,6 +89,8 @@ public boolean execute(String action, final JSONArray args, final CallbackContex
                 offError(args, callbackContext);
             } else if (action.equals("startTrip")) {
                 startTrip(args, callbackContext);
+            } else if (action.equals("updateTrip")) {
```
Comment (brettguenther) [inline src/android/src/main/java/io/radar/cordova/RadarCordovaPlugin.java:92]: i think we should expose getTripOptions + make the status value optional.

File: src/android/src/main/java/io/radar/cordova/RadarCordovaPlugin.java (hunk 1 above)
Comment (brettguenther) [inline src/android/src/main/java/io/radar/cordova/RadarCordovaPlugin.java:854]: this should be defined as matrix for RadarRouteMatrix, right?

**Review by brettguenther — COMMENTED** [review]
we also need to expose the cordova functions in Radar.js

File: src/android/src/main/java/io/radar/cordova/RadarCordovaPlugin.java (hunk 3 above)
Comment (kochis) [reply src/android/src/main/java/io/radar/cordova/RadarCordovaPlugin.java:92]: I can add this if its blocking / needed. It looks like `status` is optional below?

**Comment by tjulien** [comment]
https://app.clubhouse.io/radarlabs/story/6444/expose-gettripoptions-and-updatetrip-in-cordova

**Review by kochis — APPROVED** [review]

File: src/android/src/main/java/io/radar/cordova/RadarCordovaPlugin.java [hunk 4]
```diff
@@ -461,21 +471,90 @@ public void startTrip(final JSONArray args, final CallbackContext callbackContex
...
 
         RadarTripOptions options = RadarTripOptions.fromJson(optionsObj);
-        Radar.startTrip(options);
+        Radar.startTrip(options,new Radar.RadarTripCallback() {
+            @Override
+            public void onComplete(@NonNull Radar.RadarStatus radarStatus) {
+                if (radarStatus == Radar.RadarStatus.SUCCESS) {
+                    callbackContext.sendPluginResult(new PluginResult(PluginResult.Status.OK));
+                } else {
+                    callbackContext.sendPluginResult(new PluginResult(PluginResult.Status.ERROR,radarStatus.toString()));
```
Comment (nickpatrick) [inline src/android/src/main/java/io/radar/cordova/RadarCordovaPlugin.java:480]: @brettguenther @kochis Nit: Add a space between arguments

File: src/android/src/main/java/io/radar/cordova/RadarCordovaPlugin.java [hunk 5]
```diff
@@ -461,21 +471,90 @@ public void startTrip(final JSONArray args, final CallbackContext callbackContex
...
+              status = RadarTrip.RadarTripStatus.CANCELED;
+            }
+          }
+        }
+
+        Radar.updateTrip(options, status, new Radar.RadarTripCallback() {
+            @Override
+            public void onComplete(@NonNull Radar.RadarStatus radarStatus) {
+                if (radarStatus == Radar.RadarStatus.SUCCESS) {
+                    callbackContext.sendPluginResult(new PluginResult(PluginResult.Status.OK,radarStatus.toString()));
```
Comment (nickpatrick) [inline src/android/src/main/java/io/radar/cordova/RadarCordovaPlugin.java:526]: @brettguenther @kochis Nit: Add a space between arguments

(lots of these, applies everywhere)

File: src/ios/CDVRadar.m [hunk 6]
```diff
@@ -318,19 +318,76 @@ - (void)startTrip:(CDVInvokedUrlCommand *)command {
...
     [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
 }
 
+- (void)completeTrip:(CDVInvokedUrlCommand *)command {
+    [Radar completeTripWithCompletionHandler:^(RadarStatus radarStatus) {
+        if (radarStatus == RadarStatusSuccess) {
+            CDVPluginResult* pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsString:[Radar stringForStatus:radarStatus]];
+            [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
+        } else {
+            CDVPluginResult* pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR  messageAsString:[Radar stringForStatus:radarStatus]];
```
Comment (nickpatrick) [inline src/ios/CDVRadar.m:385]: Nit: Extra space before `messageAsString:`

File: src/android/src/main/java/io/radar/cordova/RadarCordovaPlugin.java (hunk 4 above)
Comment (nickpatrick) [reply src/android/src/main/java/io/radar/cordova/RadarCordovaPlugin.java:480]: We should also always use `PluginResult.Status.OK` and return `status` in a dict/obj like other endpoints for consistency. I'll make that change.

**Pull request merged**